internal/command/command.go

```diff
@@ -189,6 +192,40 @@ func RunCommand(c *Command, ctx context.Context) error {
 	return c.execute(cmdContext)
 }
 
+func createContainerForLanguage(ctx context.Context) (*commandState, error) {
```

Comment:
Maybe createCommandStateForLanguage? (But again, happy to change later potentially.)

And I'm assuming *this* is how we end up avoiding duplication - because most commands should end up being able to use this.